### DISCUSSION TITLE
Show preloader while app is loading

### DISCUSSION THREAD

**Discussion by d1i1m1o1n** [nuxt/nuxt#17083] [Questions] [answered]
I have an application with the `ssr: false` setting.
How to display the preloader until the browser has loaded all the scripts?

**Comment by Flanker72** [comment]
Seems you may use some [application runtime hooks](https://v3.nuxtjs.org/api/advanced/hooks) to track this event.
Have a look at `app:mounted`, `app:rendered` or `app:suspense:resolve` events.

**Reply by d1i1m1o1n** [reply]
At this point, all scripts will already be loaded.
I need first html appearance event. I hope this is possible...

**Reply by Flanker72** [reply]
Maybe you may use `bodyAttrs` or `Html`/`Body` meta tags to add initial classes and use hooks to remove them. So, when the browser gets the initial empty markup, you may show the preloader via CSS. Or as an another approach, you may show the preloader by default and hide it if special class appears (and add this class via hooks or after app mounted).

**Reply by d1i1m1o1n** [reply]
That's the question, how to display something before the application loads?

**Comment by d1i1m1o1n** [comment, marked as answer]
I found a solution!

Create file `server/plugins/render.ts`:

```
export default defineNitroPlugin((nitroApp) => {
    nitroApp.hooks.hook('render:html', (html, { event }) => {
        html.bodyAppend.push('<div>Loading...</div>');
    });
});
```
